image-builder/image_builder.sh

```diff
@@ -38,299 +56,373 @@ Options:
 	-r Free space of the root partition in MB ENV: ROOT_FREE_SPACE
 
 Extra environment variables:
-	AGENT_BIN:  use it to change the expected agent binary name
-	AGENT_INIT: use kata agent as init process
-	DAX: If 'yes' will build the image with DAX support. The first 2 MB of the
-	     resulting image are reserved for the device namespace information
-	     (metadata) that is used by the guest kernel to enable DAX.
+	AGENT_BIN:  Use it to change the expected agent binary name
+	AGENT_INIT: Use kata agent as init process
+	FS_TYPE:    Filesystem type to use. Only xfs and ext4 are supported.
 	USE_DOCKER: If set will build image in a Docker Container (requries docker)
 	            DEFAULT: not set
 
 
-	When DAX is 'yes', the following diagram shows how a 128M image will looks like:
-		 .-----------------------------------.
-		 |-- 2 MB --|-------- 126 MB --------|
-		 | Metadata | Rootfs (/bin,/usr,etc) |
-		 '-----------------------------------'
+Following diagram shows how the resulting image will look like
 
-		 The resulting image can be mounted if the offset of 2 MB is specified:
-		 $ sudo losetup -v -fP -o $((2*1024*1024)) kata-containers.img
+	.-----------.----------.---------------.-----------.
+	| 0 - 512 B | 4 - 8 Kb |  2M - 2M+512B |    3M     |
+	|-----------+----------+---------------+-----------+
+	|   MBR #1  |   DAX    |    MBR #2     |  Rootfs   |
+	'-----------'----------'---------------'-----------+
+	      |          |      ^      |        ^
+	      |          '-data-'      '--------'
+	      |                                 |
+	      '--------rootfs-partition---------'
 
-EOT
-exit "${error}"
-}
 
-# Maximum allowed size in MB for root disk
-MAX_IMG_SIZE_MB=2048
+MBR: Master boot record.
+DAX: Metadata required by the NVDIMM driver to enable DAX in the guest [1][2] (struct nd_pfn_sb).
+Rootfs: partition that contains the root filesystem (/usr, /bin, ect).
 
-FS_TYPE=${FS_TYPE:-"ext4"}
+Kernels and hypervisors that support DAX/NVDIMM read the MBR #2, otherwise MBR #1 is read.
```

Comment:
This is qemu specific or  kernel DAX/NVDIMM ? Imagine the kernel ?

Reply:
kernel and hypervisor must support NVDIMM to enable DAX, otherwise the first MBR is read like a normal block device